aws/resource_aws_vpc_peering_connection_options.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsVpcPeeringConnectionOptions() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsVpcPeeringConnectionOptionsCreate,
+		Read:   resourceAwsVpcPeeringConnectionOptionsRead,
+		Update: resourceAwsVpcPeeringConnectionOptionsUpdate,
+		Delete: resourceAwsVpcPeeringConnectionOptionsDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vpc_peering_connection_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"accepter":  vpcPeeringConnectionOptionsSchema(),
+			"requester": vpcPeeringConnectionOptionsSchema(),
+		},
+	}
+}
+
+func resourceAwsVpcPeeringConnectionOptionsCreate(d *schema.ResourceData, meta interface{}) error {
+	d.SetId(d.Get("vpc_peering_connection_id").(string))
+	return resourceAwsVpcPeeringConnectionOptionsUpdate(d, meta)
+}
+
+func resourceAwsVpcPeeringConnectionOptionsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	pcRaw, _, err := vpcPeeringConnectionRefreshState(conn, d.Id())()
+	if err != nil {
+		return fmt.Errorf("Error reading VPC Peering Connection: %s", err.Error())
+	}
+
+	if pcRaw == nil {
+		log.Printf("[WARN] VPC Peering Connection (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	pc := pcRaw.(*ec2.VpcPeeringConnection)
+
+	d.Set("vpc_peering_connection_id", pc.VpcPeeringConnectionId)
+
+	if pc.AccepterVpcInfo.PeeringOptions != nil {
```

Comment:
We should have `nil` checks for `pc.AccepterVpcInfo` and `pc.RequesterVpcInfo` to prevent crashes.

Reply:
OK.